File: api/SistemaPenal/Entities/Abstractions/Atividade.cs
using SistemaPenal.Interfaces;

namespace SistemaPenal.Entities.Abstractions;

public abstract class Atividade : IEntity
{
    public Guid Id { get; set; }
    public DateTime Data { get; set; }
    public Guid PrisioneiroId { get; set; }

    public Prisioneiro Prisioneiro { get; set; } = null!;
}

File: sistema-penal/src/models/abstract/Atividade.ts
import type { IEntity } from "../interface/IEntity";
import type { Prisioneiro } from "../Prisioneiro";

export abstract class Atividade implements IEntity {
  id: string;
  data: Date;
  prisioneiroId: string;
  prisioneiro: Prisioneiro;

  constructor(
    id: string,
    data: Date,
    prisioneiroId: string,
    prisioneiro: Prisioneiro,
  ) {
    this.id = id;
    this.data = data;
    this.prisioneiroId = prisioneiroId;
    this.prisioneiro = prisioneiro;
  }
}
